functests/sctp/sctp.go

```diff
@@ -70,6 +79,16 @@ var _ = Describe("sctp", func() {
 
 		err = namespaces.Clean(testNamespace, client.Client)
 		Expect(err).ToNot(HaveOccurred())
+
+		if sctpNodeSelector == "" {
```

Comment:
Check the `roleWorkerCNF` variable here, if set use it for the selector.
Otherwise, do the discovery and fail if it returns an error with `Expect`